app/components/toolbarSearch/toolbarSearchDirective.js

```diff
@@ -120,11 +125,19 @@
 
                 function listenSuggestWords() {
                     vm.$on('setSuggestWords', function(event, dataRawKeywords) {
+                        vm.tagSwitch.disable = false;
                         vm.suggestedKeywords = dataRawKeywords.filter(function(obj) {
                             return obj.value.length >= 3;
                         });
                     });
                 }
+
+                function toggleSuggestKeywords() {
+                    var numberKeywords = vm.numberKeywords || 5;
+                    vm.filter.textLimit = vm.tagSwitch.value ? numberKeywords : null;
+                    vm.tagSwitch.disable = true;
```

Comment:
it's performance is not good - it takes a few seconds to get the list of keywords back to the user